parsers/KPX.py

```diff
@@ -232,79 +182,109 @@ def get_long_term_prod_data(
         # order of production_values
         # 0. other, 1. gas, 2. renewable, 3. coal, 4. nuclear
         # other can be negative as well as positive due to pumped hydro
+        production_mix = ProductionMix()
+        production_mix.add_value("unknown", production_values[0] + production_values[2])
+        production_mix.add_value("gas", production_values[1])
+        production_mix.add_value("coal", production_values[3])
+        production_mix.add_value("nuclear", production_values[4])
+        production_list.append(
+            zoneKey=zone_key,
+            datetime=dt,
+            source=KR_SOURCE,
+            production=production_mix,
+        )
+    return production_list
 
-        data["datetime"] = arw_datetime
-        data["production"]["unknown"] = production_values[0] + production_values[2]
-        data["production"]["gas"] = production_values[1]
-        data["production"]["coal"] = production_values[3]
-        data["production"]["nuclear"] = production_values[4]
 
-        all_data.append(data)
+def get_real_time_prod_data(
+    zone_key: ZoneKey = ZoneKey("KR"),
+    session: Session = Session(),
+    logger: Logger = getLogger(__name__),
+) -> ProductionBreakdownList:
+    res = session.get(REAL_TIME_URL, verify=False)
 
-    return all_data
+    production_list = ProductionBreakdownList(logger)
 
+    # Extract object with data
+    data_source = re.search(r"var ictArr = (\[\{.+\}\]);", res.text).group(1)
+    # Un-quoted keys ({key:"value"}) are valid JavaScript but not valid JSON (which requires {"key":"value"}).
+    # Will break if other keys than these are introduced. Alternatively, use a JSON5 library (JSON5 allows un-quoted keys)
+    data_source = re.sub(
+        r'"(localCoal|newRenewable|oil|once|gas|nuclearPower|coal|regDate|raisingWater|waterPower|seq)"',
+        r'"\1"',
+        data_source,
+    )
+    json_obj = json.loads(data_source)
 
-def get_granular_real_time_prod_data(session: Optional[Session] = None) -> dict:
-    r0 = session or Session()
-    res_0 = r0.get(REAL_TIME_URL, verify=False)
-    chart_data = extract_chart_data(res_0.text)
+    for item in json_obj:
+        if item["regDate"] == "0":
+            break
 
-    return chart_data
+        dt = TIMEZONE.localize(datetime.strptime(item["regDate"], "%Y-%m-%d %H:%M"))
+
+        production_mix = ProductionMix()
```

Comment:
```suggestion
        production_mix = ProductionMix(
           coal=float(item["coal"]) + float(item["localCoal"]),
           gas=float(item["gas"]),
           hydro=float(item["waterPower"]),
           nuclear=float(item["nuclearPower"]),
           oil=float(item["oil"]),
           unknown=float(item["newRenewable"])
        )
```
You can also initialize the values like this. I think it's a bit more compact but up to you :)
Also @VIKTORVAV99  recently added a default rounding so that all values across all parsers are rounded by the same method, so no need for this round anymore.